js_modules/dagster-ui/packages/ui-core/src/asset-data/AssetHealthDataProvider.tsx

```diff
@@ -35,29 +35,26 @@ function init() {
           },
         });
       } else {
-        healthResponse = {data: {assetNodes: [] as AssetHealthFragment[]}};
+        healthResponse = {
+          data: {
+            assetsOrError: {nodes: [] as AssetHealthFragment[], __typename: 'AssetConnection'},
+          },
+        };
       }
 
       const {data} = healthResponse;
+      switch (data.assetsOrError.__typename) {
+        case 'PythonError':
+          throw new Error('Python error');
+        case 'AssetConnection':
+          break;
+        default:
+          throw new Error('Unknown error');
+      }
 
       const result: Record<string, AssetHealthFragment> = Object.fromEntries(
-        data.assetNodes.map((node) => [tokenForAssetKey(node.assetKey), node]),
+        data.assetsOrError.nodes.map((node) => [tokenForAssetKey(node.key), node]),
       );
-
-      // External assets are not included in the health response, so as a workaround we add them with a null assetHealth
-      keys.forEach((key) => {
-        if (!result[key]) {
-          result[key] = {
-            __typename: 'AssetNode',
-            assetKey: {
-              __typename: 'AssetKey',
-              ...tokenToAssetKey(key),
-            },
-            assetMaterializations: [],
-            assetHealth: null,
-          };
-        }
-      });
       return result;
```

Comment:
Is there a case where a key passed into the resolver doesn't return a result? If we're not certain would be nice to keep this as a work-around otherwise fine with deleting this

Reply:
it should always return a result for an asset key, even if that key doesn't exist in the DB, but not harmful to keep this as a backup. we may change the resolver to only return assets that are in the DB in the future